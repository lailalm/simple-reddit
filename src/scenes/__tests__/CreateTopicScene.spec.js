/* eslint-disable react/jsx-closing-tag-location */
import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';

import ConnectedCreateTopicScene, { CreateTopicScene } from 'scenes/CreateTopicScene';

import {
  createNewTopic,
} from 'actions/TopicAction';

import {
  CREATE_NEW_TOPIC,
} from 'utils/ActionTypes';

const mockNavigation = {
  navigate: jest.fn(),
  setParams: jest.fn(),
};

// Snapshot for CreateTopicScene React Component
describe('>>> CreateTopicScene --- Snapshot', () => {
  it('+++ capturing Snapshot of CreateTopicScene', () => {
    const renderedValue = renderer.create(<CreateTopicScene navigation={mockNavigation} />)
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('>>> CreateTopicScene --- Shallow Render', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CreateTopicScene navigation={mockNavigation} />);
  });

  it('+++ render the dummy component', () => {
    expect(wrapper.length).toEqual(1);
  });
});

describe('>>> CreateTopicScene --- React-Redux (Mount + wrapping in <Provider>)', () => {
  const origConsole = console.error;
  const initialState = {};
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    console.error = () => {};
    store = mockStore(initialState);
    wrapper = mount(<Provider store={store} >
      <ConnectedCreateTopicScene navigation={mockNavigation} />
    </Provider>);
  });

  // HACK: There isn't any react native adapter, so I'm using the react adapter.
  //       This will temporarily silence propType warnings
  afterEach(() => {
    console.error = origConsole;
  });

  it('+++ render the connected component', () => {
    expect(wrapper.find(ConnectedCreateTopicScene).length).toEqual(1);
  });

  it('+++ check action on dispatching ', () => {
    store.dispatch(createNewTopic('title of the topic'));
    const action = store.getActions();
    expect(action[0].type).toBe(CREATE_NEW_TOPIC);
  });
});
