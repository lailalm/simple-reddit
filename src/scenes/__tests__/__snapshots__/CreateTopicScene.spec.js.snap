// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> CreateTopicScene --- Snapshot +++ capturing Snapshot of CreateTopicScene 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    maxLength={255}
    multiline={true}
    numberOfLines={10}
    onChangeText={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 300,
        "margin": 10,
        "padding": 10,
      }
    }
    value=""
  />
</View>
`;
